### PR TITLE
Add min_site_addition_per_year parameter to downscaling

### PR DESCRIPTION
**Pull request by gabezuckerman**
Adds a minimum threshold for per-site load additions during downscaling. After bootstrapping and calibration, sites receiving less than the minimum are zeroed out and their load is redistributed to remaining sites. When all sites fall below the threshold, load is concentrated onto the highest-priority sites.

Closes #41

- New config parameter: [`min_site_addition_per_year`](optional, [float > 0]
- Post-calibration iterative redistribution loop
- Fallback: concentrates load onto top-N sites when all are undersized
- Negative projection year handling
- Non-negative capacity clamping
- Tests parametrized with [`min_site_addition_per_year`] in [1, None]

PS I did this with Claude so let me know if there are any issues...

### CODE REVIEW

File: reVeal/load.py [hunk 1]
```diff
@@ -334,20 +381,92 @@ def downscale_total(
...
+                    LOGGER.warning(
+                        "Downscaled load for %d sites exceeds the incremental "
+                        "developable capacity in year %d due to "
+                        "min_site_addition_per_year redistribution.",
+                        overbuilt.sum(),
+                        year,
+                    )
+                else:
+                    raise ValueError(
+                        f"Downscaled load for {overbuilt.sum} sites exceeds the "
```
[the remark this answers is not in the document]
Comment (ppinchuk) [reply reVeal/load.py]: This is worth addressing

File: reVeal/load.py [hunk 2]
```diff
@@ -290,11 +299,49 @@ def downscale_total(
...
+                    if current_sum > 0:
+                        grid_year_df[f"new_{load_value_col}"] = (
+                            load_projected_in_year
+                            * grid_year_df[current_total_col]
+                            / current_sum
+                        )
+                        # Ensure no site's total drops below zero
+                        grid_year_df[f"new_{load_value_col}"] = grid_year_df[
+                            f"new_{load_value_col}"
+                        ].clip(lower=-grid_year_df[current_total_col])
```
[the remark this answers is not in the document]
Comment (ppinchuk) [reply reVeal/load.py:326]: This can just be boiled down to "how do we handle cases where `load_projected_in_year` decreases by more than the total capacity?"

For example, if you have 70 MW of total capacity and the load decreases by 100 MW. @gabezuckerman is this even a realistic scenario? Should we just error in those cases? Or do you expect to see something like this in the wild? If the latter, then let's add a test explicitly checking that all sites get zeroed out in this scenario.

File: reVeal/load.py [hunk 3]
```diff
@@ -334,20 +381,92 @@ def downscale_total(
...
+                        LOGGER.warning(
+                            "All sites in year %d fall below "
+                            "min_site_addition_per_year (%.2f per year, "
+                            "%.2f for this time step). Concentrating load "
+                            "onto %d sites.",
+                            year,
+                            min_site_addition_per_year,
+                            min_inc,
+                            n_sites,
+                        )
```
[the remark this answers is not in the document]
Comment (ppinchuk) [reply reVeal/load.py:427]: Yea, another interesting edge case that would be worth thinking about and then adding a test around

File: reVeal/load.py [hunk 4]
```diff
@@ -334,20 +381,92 @@ def downscale_total(
...
-                    f"developable capacity in year {year}."
-                )
+                if min_site_addition_per_year:
+                    LOGGER.warning(
+                        "Downscaled load for %d sites exceeds the incremental "
+                        "developable capacity in year %d due to "
+                        "min_site_addition_per_year redistribution.",
+                        overbuilt.sum(),
+                        year,
+                    )
```
Comment (ppinchuk) [inline reVeal/load.py:494]: So `min_site_addition_per_year` is essentially the "stronger" requirement compared to `max_site_addition_per_year`, right? Specifically a site can get more than `max_site_addition_per_year` because of load inherited from sites with less than `min_site_addition_per_year`. Is this appropriate behavior? If so, can we add this tidbit to the documentation since it's non-obvious?

Comment (ppinchuk) [reply reVeal/load.py:494]: P.S. Copilot gave a more complex option for this problem - re-distribute to non-overbuilt sites. Doable but more complex

**Review by ppinchuk — COMMENTED** [review]
Some questions/comments

File: reVeal/load.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (gabezuckerman) [reply reVeal/load.py:326]: @copilot apply changes based on [this feedback](https://github.com/NatLabRockies/reVeal/pull/48#discussion_r3030597056). When distributing the remaining negative load, put as much as possible in the least suitable site, before moving to the next least suitable site. Add tests to ensure that sites get zeroed out appropriately.

File: tests/test_load.py [hunk 5]
```diff
@@ -109,14 +111,27 @@ def test_downscale_total(data_dir, max_site_addition_per_year, reduce_output):
...
+            year_df = results_df[results_df["year"] == year]
+            new_load = year_df["new_dc_load_mw"]
+            violators = new_load[(new_load > 0) & (new_load < min_site_addition_per_year)]
+            assert len(violators) == 0, (
+                f"Year {year}: {len(violators)} sites have new load below "
+                f"min_site_addition_per_year ({min_site_addition_per_year})"
+            )
+    else:
+        expected_df = gpd.read_file(expected_src)
+        assert_geodataframe_equal(results_df, expected_df, check_like=True)
```
[the remark this answers is not in the document]
Comment (gabezuckerman) [reply tests/test_load.py:128]: @copilot apply changes based on [this feedback](https://github.com/NatLabRockies/reVeal/pull/48#discussion_r3030597109)

File: reVeal/load.py (hunk 4 above)
Comment (gabezuckerman) [reply reVeal/load.py:494]: I asked copilot to fix this by redistributing to non-overbuilt sites. I think both should be hard constraints and there should be enough sites so that both can be met

File: reVeal/load.py [hunk 6]
```diff
@@ -290,11 +299,49 @@ def downscale_total(
...
+                    else:
+                        LOGGER.warning(
+                            "Projected load for year %d is negative (%.2f) "
+                            "but there is no existing load to decrease.",
+                            year,
+                            load_projected_in_year,
+                        )
+                    grid_year_df[current_total_col] += grid_year_df[
+                        f"new_{load_value_col}"
+                    ]
```
[the remark this answers is not in the document]
Comment (gabezuckerman) [reply reVeal/load.py:336]: @copilot apply changes based on [this feedback](https://github.com/NatLabRockies/reVeal/pull/48#discussion_r3030597070). Ensure earlier changes don't already address this.

**Review by ppinchuk — APPROVED** [review]
ok, if it's doing what you want, then we can merge it. We can do some code cleanup later down the road
